src/EditorFeatures/Core/Shared/Tagging/EventSources/TaggerEventSources.SemanticChangedEventSource.cs

```diff
@@ -48,7 +48,10 @@ private void OnSubjectBufferChanged(object sender, TextContentChangedEventArgs e
             {
                 // Whenever this subject buffer has changed, we always consider that to be a 
                 // semantic change.
-                this.RaiseChanged();
+                if (e.Changes.Any())
```

Comment:
What about content type changes? And why does this exist at all when the OnOpenedDocumentSemanticChanged looks like it goes out of it's way to not raise changes for itself?